From 015ebd06ff7408a950cd2dece8f24fcfadc8d08e Mon Sep 17 00:00:00 2001
From: Tom Powell <tom@powell.io>
Date: Wed, 30 May 2018 17:27:12 -0700
Subject: [PATCH 208/214] Drop jekyll-rss

I'd like to support two RSS feeds. jekyll-rss does not support this.
RSS is an easy format, just generate the feed ourselves.

Change-Id: I4a548275ccf9c0431b740fb9f8bbca6a037d5e9a
---
 Gemfile      |  1 -
 Gemfile.lock |  1 -
 _config.yml  |  1 -
 feed.xml     | 20 ++++++++++++++++++++
 4 files changed, 20 insertions(+), 3 deletions(-)
 create mode 100644 feed.xml

diff --git a/Gemfile b/Gemfile
index 9218999..71c440b 100644
--- a/Gemfile
+++ b/Gemfile
@@ -6,6 +6,5 @@ gem "therubyracer", "0.12.3"
 
 
 group :jekyll_plugins do
-   gem "jekyll-feed", "~> 0.6"
    gem 'github-pages', '~> 106'
 end
diff --git a/Gemfile.lock b/Gemfile.lock
index 9081a8f..9063f79 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -149,7 +149,6 @@ PLATFORMS
 DEPENDENCIES
   github-pages (~> 106)
   jekyll (= 3.3.1)
-  jekyll-feed (~> 0.6)
   therubyracer (= 0.12.3)
 
 BUNDLED WITH
diff --git a/_config.yml b/_config.yml
index 5a6a59d..29b5503 100644
--- a/_config.yml
+++ b/_config.yml
@@ -74,7 +74,6 @@ sass:
 # Use the following plug-ins
 gems:
   - jekyll-sitemap # Create a sitemap using the official Jekyll sitemap gem
-  - jekyll-feed # Create an Atom feed using the official Jekyll feed gem
 
 # Exclude these files from your production _site
 exclude:
diff --git a/feed.xml b/feed.xml
new file mode 100644
index 0000000..a0c8628
--- /dev/null
+++ b/feed.xml
@@ -0,0 +1,20 @@
+---
+layout: null
+---
+<?xml version="1.0" encoding="utf-8"?>
+<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
+  <channel>
+    <title>LineageOS</title>
+    <description>LineageOS Android Dstribution - Blog</description>
+    <link>{{ site.url }}</link>
+    {% for post in site.categories.blog %}
+        <item>
+            <title>{{ post.title | xml_escape }}</title>
+            <description> {{ post.excerpt  | xml_escape }}</description>
+            <pubDate>{{ post.date | date_to_xmlschema }}</pubDate>
+            <link>{{ post.url | prepend: site.url }}</link>
+            <guid isPermaLink="true">{{ post.url | prepend: site.url }}</guid>
+        </item>
+    {% endfor %}
+  </channel>
+</rss>
-- 
2.17.1

